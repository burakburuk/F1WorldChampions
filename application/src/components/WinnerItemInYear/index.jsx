import React from 'react';
import PropTypes from 'prop-types';
import Skeleton from 'react-loading-skeleton';
import { TableRow, TableCell } from '../Table';

const WinnerItemInYear = (props) => {
    const {
        name,
        race,
        company,
        date,
        highlight,
    } = props;

    return (
        <TableRow>
            <TableCell>
                {race || <Skeleton/>}
            </TableCell>
            <TableCell textColor={highlight ? '#ec7373' : '#000000'}>
                {name || <Skeleton/>}
            </TableCell>
            <TableCell>
                {date || <Skeleton/>}
            </TableCell>
            <TableCell>
                {company || <Skeleton/>}
            </TableCell>
        </TableRow>
    );
};

WinnerItemInYear.propTypes = {
    name: PropTypes.string,
    race: PropTypes.string,
    company: PropTypes.string,
    date: PropTypes.string,
    highlight: PropTypes.bool,
};

export default WinnerItemInYear;
