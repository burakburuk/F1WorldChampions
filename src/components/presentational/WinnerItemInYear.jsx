import React from 'react';
import PropTypes from 'prop-types';
import Skeleton from 'react-loading-skeleton';

const WinnerItemInYear = (props) => {
    const {
        name,
        race,
        company,
        date,
        highlight,
    } = props;

    return (
        <tr>
            <td>
                <div className="item-race">
                    {race || <Skeleton/>}
                </div>
            </td>
            <td>
                <div className={highlight ? 'by-name-item-highlight' : 'by-driver'}>
                    {name || <Skeleton/>}
                </div>
            </td>
            <td>
                <div className="by-item by-year-item item-date">
                    {date || <Skeleton/>}
                </div>
            </td>
            <td>
                <div className="by-item by-year-item item-company">
                    {company || <Skeleton/>}
                </div>
            </td>
        </tr>
    );
};

WinnerItemInYear.propTypes = {
    name: PropTypes.string,
    race: PropTypes.string,
    company: PropTypes.string,
    date: PropTypes.string,
    highlight: PropTypes.bool,
};

export default WinnerItemInYear;
