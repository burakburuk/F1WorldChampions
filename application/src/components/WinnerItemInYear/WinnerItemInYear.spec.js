import React from 'react';
import { shallow } from 'enzyme';
import WinnerItemInYear from '.';

it('should match with snapshot', () => {
    const wrapper = shallow(<WinnerItemInYear />);
    expect(wrapper).toMatchSnapshot();
});
