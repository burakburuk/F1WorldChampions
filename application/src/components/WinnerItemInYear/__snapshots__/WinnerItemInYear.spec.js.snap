// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match with snapshot 1`] = `
<TableRow>
  <TableCell>
    <Skeleton
      count={1}
      duration={1.2}
      width={null}
      wrapper={null}
    />
  </TableCell>
  <TableCell
    textColor="#000000"
  >
    <Skeleton
      count={1}
      duration={1.2}
      width={null}
      wrapper={null}
    />
  </TableCell>
  <TableCell>
    <Skeleton
      count={1}
      duration={1.2}
      width={null}
      wrapper={null}
    />
  </TableCell>
  <TableCell>
    <Skeleton
      count={1}
      duration={1.2}
      width={null}
      wrapper={null}
    />
  </TableCell>
</TableRow>
`;
